bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java

```diff
@@ -519,8 +545,11 @@ private void publishChannel(ChannelUID channelUID) {
 
     private @Nullable State updateCurrentTemp() throws GreeException {
         if (device.hasStatusValChanged(GREE_PROP_CURRENT_TEMP_SENSOR)) {
-            return new DecimalType(device.getIntStatusVal(GREE_PROP_CURRENT_TEMP_SENSOR) + INTERNAL_TEMP_SENSOR_OFFSET
-                    + config.currentTemperatureOffset.doubleValue());
+            double temp = device.getIntStatusVal(GREE_PROP_CURRENT_TEMP_SENSOR);
+            return temp != 0
```

Comment:
Just hope it will never gets 40 degrees C :wink:

Reply:
The reported temp can never be 0. This would be -40°C, I think there is some room for the implementation :-)

Reply:
Ah you're right I read that the wrong way..